### PR TITLE
Update CHANGELOG.md

### PR DESCRIPTION
**Pull request by anwittin**

### CODE REVIEW

**Review by tomcf-hcp — APPROVED** [review]
Looks good to me

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,245 @@
 - [v1.0.0 - v1.9.10](CHANGELOG-pre-v1.10.md)
 - [v0.11.6 and earlier](CHANGELOG-v0.md)
 
+## 1.17.0-RC1
+### May 30, 2024
+CHANGES:
```
Comment (mladlow) [inline CHANGELOG.md:7]:
```suggestion

CHANGES:

```
I think you'll need a space here for this to render correctly, and for the rest of the headings in the RC1 entry.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,245 @@
 - [v1.0.0 - v1.9.10](CHANGELOG-pre-v1.10.md)
 - [v0.11.6 and earlier](CHANGELOG-v0.md)
 
+## 1.17.0-RC1
```
Comment (mladlow) [inline CHANGELOG.md:5]:
```suggestion
## 1.17.0-rc1
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,245 @@
...
+* secrets/pki (enterprise): cert_role parameter within authenticators.cert EST configuration handler could not be set
+* secrets/transit: Use 'hash_algorithm' parameter if present in HMAC verify requests. Otherwise fall back to deprecated 'algorithm' parameter. [[GH-27211](https://github.com/hashicorp/vault/pull/27211)]
+* ui: Fix KVv2 cursor jumping inside json editor after initial input. [[GH-27120](https://github.com/hashicorp/vault/pull/27120)]
+* ui: Fix KVv2 json editor to allow null values. [[GH-27094](https://github.com/hashicorp/vault/pull/27094)]
+* ui: Fix broken help link in console for the web command. [[GH-26858](https://github.com/hashicorp/vault/pull/26858)]
+* ui: Fix link to v2 generic secrets engine from secrets list page. [[GH-27019](https://github.com/hashicorp/vault/pull/27019)]
+* ui: Prevent perpetual loading screen when Vault needs initialization [[GH-26985](https://github.com/hashicorp/vault/pull/26985)]
+* ui: Refresh model within a namespace on the Secrets Sync overview page. [[GH-26790](https://github.com/hashicorp/vault/pull/26790)]
+
+## 1.15.9 Enterprise
```
Comment (mladlow) [inline CHANGELOG.md:210]: Could you move this section to between 1.15.8 and 1.16.0, and the 1.14 addition to between 1.14.11 and 1.15.0 for consistency with prior changelogs?

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,245 @@
...
+* secrets/pki: sign-intermediate API will truncate notAfter if calculated to go beyond the signing issuer's notAfter. Previously the notAfter was permitted to go beyond leading to invalid chains. [[GH-26796](https://github.com/hashicorp/vault/pull/26796)]
+* secrets/terraform: Update plugin to v0.8.0 [[GH-27147](https://github.com/hashicorp/vault/pull/27147)]
+* ui: Update dependencies including D3 libraries [[GH-26346](https://github.com/hashicorp/vault/pull/26346)]
+* ui: Upgrade Ember data from 4.11.3 to 4.12.4 [[GH-25272](https://github.com/hashicorp/vault/pull/25272)]
+* ui: Upgrade Ember to version 5.4 [[GH-26708](https://github.com/hashicorp/vault/pull/26708)]
+* ui: deleting a nested secret will no longer redirect you to the nearest path segment [[GH-26845](https://github.com/hashicorp/vault/pull/26845)]
+* ui: flash messages render on right side of page [[GH-25459](https://github.com/hashicorp/vault/pull/25459)]
+
+FEATURES:
+* **PKI Certificate Metadata (enterprise)**:  Add Certificate Metadata Functionality to Record and Return Client Information about a Certificate.
```
Comment (mladlow) [inline CHANGELOG.md:53]: It's great to see these all visually differentiated for the changelog!

**Review by mladlow — APPROVED** [review]

**Pull request merged**